Zend/zend_language_parser.y

```diff
@@ -592,6 +595,33 @@ interface_declaration_statement:
 			{ $$ = zend_ast_create_decl(ZEND_AST_CLASS, ZEND_ACC_INTERFACE, $<num>2, $5, zend_ast_get_str($3), NULL, $4, $7, NULL, NULL); }
 ;
 
+enum_declaration_statement:
+		T_ENUM { $<num>$ = CG(zend_lineno); }
+		T_STRING enum_primitive_type implements_list backup_doc_comment '{' class_statement_list '}'
+			{ $$ = zend_ast_create_decl(ZEND_AST_CLASS, ZEND_ACC_ENUM|ZEND_ACC_FINAL, $<num>2, $6, zend_ast_get_str($3), NULL, $5, $8, NULL, $4); }
+;
+
+enum_case:
+		T_CASE identifier enum_case_expr ';'
+			{ $$ = zend_ast_create(ZEND_AST_ENUM_CASE, $2, $3, NULL); }
+;
+
+enum_case_expr:
+		%empty	{ $$ = NULL; }
+	|	'=' enum_case_value { $$ = $2; }
+;
+
+enum_case_value:
+		T_LNUMBER 	{ $$ = $1; }
+	|	T_DNUMBER 	{ $$ = $1; }
```

Comment:
T_DNUMBER is used for IS_DOUBLE, but only IS_LONG and IS_STRING are allowed. Is there a reason to include this?